frontend/test/actions/RegistrationUriTest.scala

```diff
@@ -0,0 +1,55 @@
+
+  "from request" should {
+    "contain correct identity tracking code for home page" in {
+      assertCodeFor("MEM_HOME_SUP", "/join/supporter/enter-details", "http://mem.thegulocal.com/")
+      assertCodeFor("MEM_HOME_PAR", "/join/partner/enter-details", "http://mem.thegulocal.com/")
+      assertCodeFor("MEM_HOME_PAT", "/join/patron/enter-details", "http://mem.thegulocal.com/")
+      assertCodeFor("MEM_HOME_FRI", "/join/friend/enter-details", "http://mem.thegulocal.com/")
+    }
+
+    "contain correct identity tracking code for events page" in {
+      assertCodeFor("MEM_EVT_SUP", "/join/supporter/enter-details", "https://membership.theguardian.com/events")
+      assertCodeFor("MEM_EVT_PAR", "/join/partner/enter-details", "https://membership.theguardian.com/events")
+      assertCodeFor("MEM_EVT_PAT", "/join/patron/enter-details", "https://membership.theguardian.com/events")
+      assertCodeFor("MEM_EVT_FRI", "/join/friend/enter-details", "https://membership.theguardian.com/events")
+    }
+
+    "contain correct identity tracking code for event details page" in {
+      assertCodeFor("MEM_COMP_SUP", "/join/supporter/enter-details", "https://membership.theguardian.com/offers-competitions")
+      assertCodeFor("MEM_COMP_PAR", "/join/partner/enter-details", "https://membership.theguardian.com/offers-competitions")
+      assertCodeFor("MEM_COMP_PAT", "/join/patron/enter-details", "https://membership.theguardian.com/offers-competitions")
+      assertCodeFor("MEM_COMP_FRI", "/join/friend/enter-details", "https://membership.theguardian.com/offers-competitions")
+    }
+
+    "contain correct identity tracking code for US supporters" in {
+      assertCodeFor("MEM_SUPUS_SUP", "/join/supporter/enter-details", "https://membership.theguardian.com/us/supporter")
+      assertCodeFor("MEM_SUPUS_PAR", "/join/partner/enter-details", "https://membership.theguardian.com/us/supporter")
+      assertCodeFor("MEM_SUPUS_PAT", "/join/patron/enter-details", "https://membership.theguardian.com/us/supporter")
+      assertCodeFor("MEM_SUPUS_FRI", "/join/friend/enter-details", "https://membership.theguardian.com/us/supporter")
+    }
+
+
+    "contain correct identity tracking code for US supporters" in {
+      assertCodeFor("MEM_SUPUK_SUP", "/join/supporter/enter-details", "https://membership.theguardian.com/supporter")
+      assertCodeFor("MEM_SUPUK_PAR", "/join/partner/enter-details", "https://membership.theguardian.com/supporter")
+      assertCodeFor("MEM_SUPUK_PAT", "/join/patron/enter-details", "https://membership.theguardian.com/supporter")
+      assertCodeFor("MEM_SUPUK_FRI", "/join/friend/enter-details", "https://membership.theguardian.com/supporter")
+    }
+
+  }
+
+
+  private def assertCodeFor(code: String, path: String, referrer: String) =  {
+    val request = new RequestBuilder().path(path).header("referer", referrer).build()._underlyingHeader()
+    val regUri: String = RegistrationUri.parse(request)
+    println(regUri)
```

Comment:
Should this still be here? If it's to provide debug information when running the tests should it give a bit more info? e.g. `println(s"Registration URI: ${regUri}")` or something like that